ORBIT/phases/install/quayside_assembly_tow/moored.py

```diff
@@ -0,0 +1,397 @@
+
+from ORBIT.core import Vessel, WetStorage
+from ORBIT.phases.install import InstallPhase
+
+from .common import TowingGroup, TurbineAssemblyLine, SubstructureAssemblyLine
+
+
+class MooredSubInstallation(InstallPhase):
+    """
+    Installation module to model the quayside assembly, tow-out and
+    installation at sea of moored substructures.
+    """
+
+    phase = "Moored Substructure Installation"
+
+    #:
+    expected_config = {
+        "support_vessel": "str",
+        "towing_vessel": "str",
+        "towing_vessel_groups": {
+            "towing_vessels": "int",
+            "stabilization_vessels": "int",
+            "num_groups": "int (optional)",
+        },
+        "substructure": {
+            "takt_time": "int | float (optional, default: 0)",
+            "towing_speed": "int | float (optional, default: 6 km/h)",
+        },
+        "site": {"depth": "m", "distance": "km"},
+        "plant": {"num_turbines": "int"},
+        "turbine": "dict",
+        "port": {
+            "sub_assembly_lines": "int (optional, default: 1)",
+            "sub_storage": "int (optional, default: inf)",
+            "turbine_assembly_cranes": "int (optional, default: 1)",
+            "assembly_storage": "int (optional, default: inf)",
+            "monthly_rate": "USD/mo (optional)",
+            "name": "str (optional)",
+        },
+    }
+
+    def __init__(self, config, weather=None, **kwargs):
+        """
+        Creates an instance of MooredSubInstallation.
+
+        Parameters
+        ----------
+        config : dict
+            Simulation specific configuration.
+        weather : np.array
+            Weather data at site.
+        """
+
+        super().__init__(weather, **kwargs)
+
+        config = self.initialize_library(config, **kwargs)
+        self.config = self.validate_config(config)
+        self.extract_defaults()
+
+        self.setup_simulation(**kwargs)
+
+    def setup_simulation(self, **kwargs):
+        """
+        Sets up simulation infrastructure.
+        - Initializes substructure production
+        - Initializes turbine assembly processes
+        - Initializes towing groups
+        """
+
+        self.distance = self.config["site"]["distance"]
+        self.num_turbines = self.config["plant"]["num_turbines"]
+
+        self.initialize_substructure_production()
+        self.initialize_turbine_assembly()
+        self.initialize_queue()
+        self.initialize_towing_groups()
+        self.initialize_support_vessel()
+
+    def initialize_substructure_production(self):
+        """
+        Initializes the production of substructures at port. The number of
+        independent assembly lines and production time associated with a
+        substructure can be configured with the following parameters:
+
+        - self.config["substructure"]["takt_time"]
+        - self.config["port"]["sub_assembly_lines"]
+        """
+
+        try:
+            storage = self.config["port"]["sub_storage"]
```

Comment:
Number of substructures that can be stored at port.

This starts to push the original boundaries of ORBIT (no supply chain modeling) but from talking with Aaron and Luisa (and even the conversation with the bench marking group this morning), it sounds like this is an important consideration for floating substructures.

I designed the module in a way that this can be easily ignored by setting 'takt_time' to 0 and leaving 'sub_storage' to it's default of infinity. In this situation, all substructures will be "manufactured" at time 0 and the assembly won't constrain the installations at all.